SDK.md

```diff
@@ -44,6 +44,68 @@ Supports CloudEvents spec milestones and ongoing development version.
 HTTP transport support (both structured and binary).
 Widely used programming language version.
 
+### Object model structure guidelines
+
+The CloudEvents specification set consists of
+
+* the base specification that defines the abstract information model and
+  associated rules for a CloudEvent,
+* a set of extensions to the base specification that define use-case
+  specific attributes and associated rules,
+* a set of event format specifications that define how a CloudEvent and  
+  its attributes are encoded in a particular wire transfer format, and
+* a set of transport bindings that define how a wire-encoded CloudEvent is
+  interoperably transported from a publisher or intermediary to a target
+  endpoint using a particular application protocol.
+
+The object model of any API projection of CloudEvents MUST reflect the
+specification architecture with the following elements:
+
+* A generic CloudEvent class/object/structure that provides access to the
+  CloudEvent attributes and enforces data type constraints and all further
+  conformance rules defined in the core CloudEvents specification. Instances
+  of this class/object/structure MUST always be in a conformant state.
+  The generic CloudEvent abstraction SHOULD provide strongly typed access
+  to all well-known attributes, and MUST also provide name-indexed access
+  to any application-specific extension attributes that the event might carry.
+  Name-indexed access MUST include well-known attributes from the core
```

Comment:
While I like this MUST because it provides consistency to the user of the SDK for all attributes (core or extensions), I wonder if forcing all SDKs to expose the core CE properties via a map is too much of a burden?

Reply:
The rationale is that if an extension gets promoted into the core, you break existing code if that extension then vanishes from the indexed access collection. So this rule is for protecting against breaking changes for that scenario -- and for the promotion of a extension into the core I assume that the extension is popular and therefore you break a lot of code.

Reply:
yep - I understand the reason for it, I'm just trying to weigh the pros/cons of actually mandating it with a MUST vs just recommending it with a SHOULD. For example, if there's an SDK that wants to ensure strict compliance to the spec to ensure maximum interop might even go so far as to ban extensions, and therefore might only expose well defined getters/setters for the spec-defined attributes. They would be in violation of this rule even though they're CE-spec compliant.  I'm also not sure how contrived this example is but it's what I'm wondering about...